willow/Dockerfile

```diff
@@ -34,39 +29,19 @@ RUN apt-get update -qq && \
     ffmpeg \
     ufraw \
     bzip2 unzip xz-utils \
-    openjdk-8-jre-headless="$JAVA_DEBIAN_VERSION" \
-    ca-certificates-java="$CA_CERTIFICATES_JAVA_VERSION" \
+    # install open-jdk and ca-certs from jessie-backports
+    && apt-get install -t jessie-backports -y --no-install-recommends openjdk-8-jre-headless ca-certificates-java \
     && apt-get clean \
-	&& rm -rf /var/lib/apt/lists/*
-
-# add a simple script that can auto-detect the appropriate JAVA_HOME value
-# based on whether the JDK or only the JRE is installed
-RUN { \
-		echo '#!/bin/sh'; \
-		echo 'set -e'; \
-		echo; \
-		echo 'dirname "$(dirname "$(readlink -f "$(which javac || which java)")")"'; \
-	} > /usr/local/bin/docker-java-home && \
-	chmod +x /usr/local/bin/docker-java-home && \
-    set -x && [ "$JAVA_HOME" = "$(docker-java-home)" ]
+	&& rm -rf /var/lib/apt/lists/* \
+	&& /var/lib/dpkg/info/ca-certificates-java.postinst configure \
+    && mkdir -p /fits/ \
```

Comment:
I see each RUN as a group of commands that is related in some manner. For example, setting up apt repositories, running apt-get update and installing a bunch of system software.

I just saw fits as a separate (unrelated) thing to do.

I also find the pattern of super-long commands kind of smelly, but I understand why it's being used. I have also observed well maintained open source images (postgres, solr) using it in their own Dockerfiles. I'd find putting every single shell command in the Dockerfile in a single RUN directive a bit extreme though.